python/lsst/pipe/base/script/transfer_from_graph.py

```diff
@@ -62,13 +62,26 @@ def transfer_from_graph(
     qgraph = QuantumGraph.loadUri(graph)
 
     # Collect output refs that could be created by this graph.
-    output_refs: set[DatasetRef] = set(qgraph.globalInitOutputRefs())
+    original_output_refs: set[DatasetRef] = set(qgraph.globalInitOutputRefs())
     for task_def in qgraph.iterTaskGraph():
         if refs := qgraph.initOutputRefs(task_def):
-            output_refs.update(refs)
+            original_output_refs.update(refs)
     for qnode in qgraph:
         for refs in qnode.quantum.outputs.values():
-            output_refs.update(refs)
+            original_output_refs.update(refs)
+
+    # Get data repository definitions from the QuantumGraph; these can have
+    # different storage classes than those in the quanta.
+    dataset_types = {dstype.name: dstype for dstype in qgraph.registryDatasetTypes()}
```

Comment:
I see this line enough I'm starting to wonder if QuantumGraph could have a method that returned the dict.

Reply:
:+1: .  I'm informally saving up a number of changes I'd like to make to QG.